src/AspNetCoreModuleV2/CommonLib/debugutil.cpp

```diff
@@ -47,31 +52,86 @@ DebugInitialize()
-    const size_t environmentVariableValueSize = 2;
-    std::wstring environmentVariableValue(environmentVariableValueSize, '\0');
+    try
+    {
+        SetDebugFlags(Environment::GetEnvironmentVariableValue(L"ASPNETCORE_MODULE_DEBUG").value_or(L"0"));
+    }
+    catch (...)
+    {
+        // ignore
+    }
 
     try
     {
-        const auto value = std::stoi(Environment::GetEnvironmentVariableValue(L"ASPNETCORE_MODULE_DEBUG").value_or(L"0"));
+        const auto debugOutputFile = Environment::GetEnvironmentVariableValue(L"ASPNETCORE_MODULE_DEBUG_FILE");
 
-        if (value >= 1) DEBUG_FLAGS_VAR |= ASPNETCORE_DEBUG_FLAG_ERROR;
-        if (value >= 2) DEBUG_FLAGS_VAR |= ASPNETCORE_DEBUG_FLAG_WARNING;
-        if (value >= 3) DEBUG_FLAGS_VAR |= ASPNETCORE_DEBUG_FLAG_INFO;
-        if (value >= 4) DEBUG_FLAGS_VAR |= ASPNETCORE_DEBUG_FLAG_CONSOLE;
+        CreateDebugLogFile(debugOutputFile.value_or(L""));
     }
     catch (...)
     {
         // ignore
     }
+}
+
+HRESULT
+DebugInitializeFromConfig(IHttpServer& pHttpServer, IHttpApplication& pHttpApplication)
+{
+    CComPtr<IAppHostElement>        pAspNetCoreElement;
+
+    const CComBSTR bstrAspNetCoreSection = L"system.webServer/aspNetCore";
+    CComBSTR bstrConfigPath = pHttpApplication.GetAppConfigPath();
+
+    RETURN_IF_FAILED(pHttpServer.GetAdminManager()->GetAdminSection(bstrAspNetCoreSection,
+        bstrConfigPath,
+        &pAspNetCoreElement));
+
+    STRU debugFile;
+    RETURN_IF_FAILED(ConfigUtility::FindDebugFile(pAspNetCoreElement, debugFile));
+
+    STRU debugValue;
+    RETURN_IF_FAILED(ConfigUtility::FindDebugLevel(pAspNetCoreElement, debugValue));
+
+    SetDebugFlags(debugFile.QueryStr());
+
+    CreateDebugLogFile(debugFile.QueryStr());
+
+    return S_OK;
+}
 
+void SetDebugFlags(const std::wstring &debugValue)
+{
     try
     {
-        const auto debugOutputFile = Environment::GetEnvironmentVariableValue(L"ASPNETCORE_MODULE_DEBUG_FILE");
+        if (!debugValue.empty())
+        {
+            const auto value = std::stoi(debugValue.c_str());
 
-        if (debugOutputFile.has_value())
+            if (value >= 1) DEBUG_FLAGS_VAR |= ASPNETCORE_DEBUG_FLAG_ERROR;
+            if (value >= 2) DEBUG_FLAGS_VAR |= ASPNETCORE_DEBUG_FLAG_WARNING;
+            if (value >= 3) DEBUG_FLAGS_VAR |= ASPNETCORE_DEBUG_FLAG_INFO;
+            if (value >= 4) DEBUG_FLAGS_VAR |= ASPNETCORE_DEBUG_FLAG_CONSOLE;
+        }
+    }
+    catch (...)
+    {
+        // ignore
+    }
+}
+
+void CreateDebugLogFile(const std::wstring &debugOutputFile)
+{
+    try
+    {
+        if (!debugOutputFile.empty())
         {
-            g_logFile = CreateFileW(debugOutputFile.value().c_str(),
+            if (g_logFile != INVALID_HANDLE_VALUE)
+            {
+                WLOG_INFOF(L"Switching debug log files to %s", debugOutputFile.c_str());
+                CloseHandle(g_logFile);
+                DEBUG_FLAGS_VAR &= ~ASPNETCORE_DEBUG_FLAG_FILE;
```

Comment:
why?

Reply:
If CreateFile fails, the handle will be invalid and we don't want to log to it.